miprometheus/utils/sampler_factory.py

```diff
@@ -0,0 +1,169 @@
+
+import os
+import logging
+import torch.utils.data.sampler
+
+
+class SamplerFactory(object):
+    """
+    Class returning sampler depending on the name provided in the \
+    list of parameters.
+    """
+
+    @staticmethod
+    def build(problem, params):
+        """
+        Static method returning particular sampler, depending on the name \
+        provided in the list of parameters.
+
+        :param problem: Instance of an object derived from the Problem class.
+        :type problem: ``problems.Problem``
+
+        :param params: Parameters used to instantiate the sampler.
+        :type params: ``utils.param_interface.ParamInterface``
+
+        ..note::
+
+            ``params`` should contains the exact (case-sensitive) class name of the sampler to instantiate.
+
+        :return: Instance of a given sampler or None if section not present (None) or coudn't build the sampler.
+        """
+        logging.basicConfig(level=logging.INFO)
+        logger = logging.getLogger('SamplerFactory')
+
+        # Check if sampler is required, i.e. 'sampler' section is empty.
+        if not params:
+            return None
+
+        try: 
+            # Check presence of the name attribute.
+            if 'name' not in params:
+                raise Exception("Sampler configuration section does not contain the key 'name'")
+
+            # Get the class name.
+            name = params['name']
+
+            # Verify that the specified class is in the controller package.
+            if name not in dir(torch.utils.data.sampler):
+                raise Exception("Could not find the specified class '{}' in the controllers package".format(name))
+
+            # Get the actual class.
+            sampler_class = getattr(torch.utils.data.sampler, name)
+
+            # Ok, proceed.
+            logger.info('Loading the {} controller from {}'.format(name, sampler_class.__module__))
+
+            # Handle "special" case.
+            if sampler_class.__name__ == 'SubsetRandomSampler':
+
+                indices = params['indices']
+
+                # Analyze the type.
+                if type(indices) == str:
+
+                    # Try to open the file.
+                    try:
+                        # from expanduser()'s doc: If the expansion fails or if the path does not begin
+                        # with a tilde, the path is returned unchanged. -> So operation below should be safe.
+                        file = open(os.path.expanduser(indices), "r")
+                        # Read the file.
+                        indices = file.readline() 
+                        # Truncate the last "enter"
+                        indices = indices[:-1]
+                        file.close()
+
+                    except Exception:
+                        # Ok, this is not a file.
+                        pass
+
+                    # If indices are already square brackets [].
+                    if indices[0] == '[' and indices[-1] == ']':
```

Comment:
The `yaml` library that we user (`PyYaml`) leaves the option to the user to specify the type of the object that we are passing. Even instances of Python classes can be constructed using the `!!python/object` tag (pulled from the [doc](https://pyyaml.org/wiki/PyYAMLDocumentation)). And a quick test shows that it does work:

![yaml_load](https://user-images.githubusercontent.com/33840936/47576027-5424ff80-d8f8-11e8-9fb6-12617bbc96e4.png)

So instead of `if indices[0] == '[' and indices[-1] == ']':`, we could use `if type(indices) == list:`, and this would simplify the code.